completed task 2

rustychump opened the pull request.

rustychump committed:
completed task 2

alesyaisachenko commented on src/test/java/TestChangingUserData.java:
```diff
@@ -0,0 +1,87 @@
+import cards.ResponseAuthUserCard;
+import io.qameta.allure.junit4.DisplayName;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
```
не используемые импорты нужно удалять.

Коммент относится ко всему коду

alesyaisachenko commented on src/test/java/TestChangingUserData.java:
```diff
@@ -0,0 +1,87 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://stellarburgers.nomoreparties.site";
```
Эта настройка у тебя есть в каждом тесте.

Логично это вынести в отдельный класс, а потом использовать наследование.

это можно было реализовать через базовый класс, в котором задать  RequestSpecification.

alesyaisachenko commented on src/test/java/TestChangingUserData.java:
```diff
@@ -0,0 +1,87 @@
+
+
+
+    private ResponseAuthUserCard getResponseAuthUserCard(CreateUserCard userCard) {
```
нужно улучшить архитектуру кода.

все методы для формирования запроса нужно вынести в отдельный класс, например UserApi и OrderApi.

в тестах уже использовать методы из этих классов.

Это нужно обязательно исправить.

alesyaisachenko commented on src/test/java/TestChangingUserData.java:
```diff
@@ -0,0 +1,87 @@
+                .header("Content-type", "application/json")
+                .body(userCard)
+                .when()
+                .post("/api/auth/login")
```
"/api/auth/login" -- такие вещи лучше определять как константы (public static final)

alesyaisachenko commented on src/test/java/TestChangingUserData.java:
```diff
@@ -0,0 +1,87 @@
+                .body(changedUserCard)
+                .when()
+                .patch("/api/auth/user")
+                .then().statusCode(200)
```
статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

rustychump committed:
1. removed unused imports
2. created an abstract class BaseTest
3. endpoints defined as constants
4. used import static org.apache.http.HttpStatus.*

rustychump committed:
changed names of constants

rustychump committed:
1. created classes OrderApi and UserApi
2. changed code architecture
